### PR TITLE
Added optional config keys to the Pushover Backend

### PR DESCRIPTION
**Pull request by tjbenator**
Example config:
```yaml
backends:
    - pushover
pushover:
    user_key: redacted                     
    sound: alien
    priority: 2
    expire: 60
    retry: 3600
    callback: https://callbackurl
```

### CODE REVIEW

**Comment by tjbenator** [comment]
Resolves issue #13

File: ntfy/backends/pushover.py [hunk 1]
```diff
@@ -17,6 +25,25 @@ def notify(title, message, user_key,
     if device:
         data['device'] = device
 
+    if sound:
+        data['sound'] = sound
+
+    if priority != 0 and priority <= 2 and priority >= -2:
```
Comment (dschep) [inline ntfy/backends/pushover.py]: Could you add an `else:` clause? Probably with something like `raise ValueError('priority must be an integer from -2 to 2')

File: ntfy/backends/pushover.py [hunk 2]
```diff
@@ -17,6 +25,25 @@ def notify(title, message, user_key,
...
+        data['priority'] = priority
+
+        # Expire, Retry, and Callback only apply to an Emergency Priority
+        if priority == 2:
+            # Retry can not be less than 30 per the API
+            if retry < 30:
+                data['retry'] = 30
```
Comment (dschep) [inline ntfy/backends/pushover.py]: Add
```
logging.getLogger(__name__).error('retry is less than 30, setting retry to 30 to comply with pushover API requirements')
```
(and at the top of file: `import logging`)

File: ntfy/backends/pushover.py [hunk 3]
```diff
@@ -17,6 +25,25 @@ def notify(title, message, user_key,
...
         data['device'] = device
 
+    if sound:
+        data['sound'] = sound
+
+    if priority != 0 and priority <= 2 and priority >= -2:
+        data['priority'] = priority
+
+        # Expire, Retry, and Callback only apply to an Emergency Priority
+        if priority == 2:
```
Comment (dschep) [inline ntfy/backends/pushover.py]: Raise errors or warn about Expire, retry & callback being ignored in an else clause if they're set.

**Comment by dschep** [comment]
Looks fantastic!. Rebase it to fix the merge conflicts and I'll :shipit:

**Comment by dschep** [comment]
And feel free to add yourself to the contributors in `README.rst` :smiley:

**Pull request merged**